common/src/main/java/com/revenuecat/purchases/common/offerings/OfferingsManager.kt

```diff
@@ -62,26 +64,37 @@ class OfferingsManager(
         backend.getOfferings(
             appUserID,
             appInBackground,
-            { offeringsJSON ->
-                offeringsFactory.createOfferings(
-                    offeringsJSON,
-                    onError = { error ->
-                        handleErrorFetchingOfferings(error, onError)
-                    },
-                    onSuccess = { offerings ->
-                        synchronized(this@OfferingsManager) {
-                            offeringsCache.cacheOfferings(offerings, offeringsJSON)
-                        }
-                        dispatch {
-                            onSuccess?.invoke(offerings)
-                        }
-                    }
-                )
-            }, { error ->
-                handleErrorFetchingOfferings(error, onError)
+            { createAndCacheOfferings(it, onError, onSuccess) },
+            { backendError ->
+                val cachedOfferingsResponse = offeringsCache.cachedOfferingsResponse
+                if (cachedOfferingsResponse == null) {
+                    handleErrorFetchingOfferings(backendError, onError)
+                } else {
+                    warnLog(OfferingStrings.ERROR_FETCHING_OFFERINGS_USING_DISK_CACHE)
```

Comment:
In iOS I'm only doing this for "server is down" errors FWIW. I don't think this is bad but might want to keep the behavior the same?

Reply:
Hmm right... I do think we can just use cached offerings for other types of errors... But we can discuss that further and be minimal for now. Will do the change 👍

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1029/commits/4270a5e86ede0f395d522a48ee8ca9ba9fab2ee7